### PR TITLE
Rem 199

### PR DESCRIPTION
**Pull request by FatimaSalami**

### CODE REVIEW

**Comment by NoraBitarSM** [comment]
functionality tested

File: src/pages/personal-settings/index.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    fullName: '',
+    languageId: '',
+    menuTheme: '',
+    languageName: ''
+  })
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+          const userData = window.sessionStorage.getItem('userData')
```
Comment (hadichahine) [inline src/pages/personal-settings/index.js:36]: Can we something regarding this? I know the team is copying this over from previous code but we need to refactor it into a utility. Maybe create a file having the session storage functions?

File: src/pages/personal-settings/index.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  })
+  
+  const postPersonalSettings = obj => {
+   
+    postRequest({
+      extension: SelfServiceRepository.SSUserInfo.set,
+      record: JSON.stringify(obj)
+    })
+      .then(res => {
+        if (res) toast.success('Record Added Successfully')
```
Comment (hadichahine) [inline src/pages/personal-settings/index.js:76]: No need for the if statement but double check with the api devs.

File: src/pages/personal-settings/index.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+  const handleSubmit = () => {
+    formik.handleSubmit()
+  }
+
+  return (
+    <>
+      <Box
+        sx={{
+          display: 'flex',
```
Comment (hadichahine) [inline src/pages/personal-settings/index.js:88]: Do we have MUI specific attributes that we can replace those with?

File: src/pages/personal-settings/index.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+            <ResourceComboBox
+              datasetId={DataSets.LANGUAGE}
+              name='languageId'
+              label={_labels.lang}
+              valueField='key'
+              displayField='value'
+              values={formik.values}
+              required     
+              onChange={(event, newValue) => {
+                formik && formik.setFieldValue('languageId', newValue?.key ?? '')
```
Comment (hadichahine) [inline src/pages/personal-settings/index.js:116]: formik should always be not null
no need for check but verify please

File: src/pages/personal-settings/index.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+              left: 0,
+              width: '100%',
+              padding: 3,
+              textAlign: 'center'
+            }}
+          >
+            <WindowToolbar onSave={handleSubmit} isSaved={true} />
+          </Grid>
+        </Grid>
+        <ErrorWindow open={errorMessage} onClose={() => setErrorMessage(null)} message={errorMessage} />
```
Comment (hadichahine) [inline src/pages/personal-settings/index.js:148]: We still want this?

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/personal-settings/index.js (hunk 3 above)
Comment (FatimaSalami) [reply src/pages/personal-settings/index.js:88]: i changed both in box and grid container at the bottom of page

File: src/pages/personal-settings/index.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+import { useResourceQuery } from 'src/hooks/resource'
+import { DataSets } from 'src/resources/DataSets'
+import * as yup from 'yup'
+import { Center } from 'devextreme-react/map'
+import { getStorageData } from 'src/storage/storageData'
+
+const PersonalSettings = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const [initialValues, setInitialValues] = useState({
```
Comment (hadichahine) [inline src/pages/personal-settings/index.js:20]: Make sure we need this mode (vs having formik.setValues)

File: src/pages/personal-settings/index.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    languageId: '',
+    menuTheme: '',
+    languageName: ''
+  })
+
+  useEffect(() => {
+    ;(async function () {
+      try {
+        const userData = getStorageData('userData')
+        const _userId = userData && userData.userId
```
Comment (hadichahine) [inline src/pages/personal-settings/index.js:37]: Do we need this check? I'm assuming if we're logged in we don't need this.

File: src/pages/personal-settings/index.js [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+          extension: SelfServiceRepository.SSUserInfo.get,
+          parameters: `_recordId=${_userId}`
+        })
+
+        setInitialValues(res.record)
+      } catch (exception) {}
+    })()
+  }, [])
+
+  const { labels: _labels } = useResourceQuery({
```
Comment (hadichahine) [inline src/pages/personal-settings/index.js:49]: There's another function ig inside resource.js that fetches only labels and access. btw access is not being used here at all.

File: src/pages/personal-settings/index.js [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    datasetId: ResourceIds.PersonalSettings
+  })
+
+  const formik = useFormik({
```
Comment (hadichahine) [inline src/pages/personal-settings/index.js:53]: useForm

File: src/pages/personal-settings/index.js [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    toast.success('Record Added Successfully')
+  }
+
+  const handleSubmit = () => {
+    formik.handleSubmit()
+  }
+
+  return (
+    <>
+      <Box display='flex' height='100%' marginTop='10px' flexDirection='column'>
```
Comment (hadichahine) [inline src/pages/personal-settings/index.js:80]: Try ommiting those.

File: src/providers/TabsContext.js
Comment (hadichahine) [inline src/providers/TabsContext.js:1]: what is this for?

**Review by hadichahine — APPROVED** [review]

**Pull request merged**